datasets.py

```diff
@@ -0,0 +1,64 @@
+from random import Random
+import ipaddr
+import datetime as dt
+
+from interface import Entry
+
+LOWEST_IP = 2 ** 30  # 64.0.0.0
+HIGHEST_IP = 2 ** 31  # 128.0.0.0
+LOWEST_TIME = dt.datetime(2001, 1, 1).toordinal()
+HIGHEST_TIME = dt.datetime(2016, 1, 1).toordinal()
+PROTOCOLS = ['TCP', 'UDP', 'DNS', 'HTTP', 'SSH', 'FTP']
+
+rnd = Random(0)
+
+
+def _generate_dataset(number_of_ips, number_of_entries_per_ip):
+    for ip_num in (rnd.randint(LOWEST_IP, HIGHEST_IP) for _ in range(number_of_ips)):
+        for _ in range(number_of_entries_per_ip):
+            yield Entry(str(ipaddr.IPAddress(ip_num)), protocol=rnd.choice(PROTOCOLS),
+                        timestamp=dt.datetime.fromordinal(rnd.randint(LOWEST_TIME, HIGHEST_TIME)))
+
+
+def s1wide():
```

Comment:
Functions could use nicer names :)

Reply:
Not my code - from the exercise documentation